lib/web_ui/test/engine/platform_dispatcher/view_focus_binding_test.dart

```diff
@@ -86,19 +86,15 @@ void testMain() {
       focusableViewElement1.focus();
       focusableViewElement2.focus();
 
-      expect(viewFocusEvents, hasLength(3));
+      expect(viewFocusEvents, hasLength(2));
 
       expect(viewFocusEvents[0].viewId, view1.viewId);
       expect(viewFocusEvents[0].state, ui.ViewFocusState.focused);
       expect(viewFocusEvents[0].direction, ui.ViewFocusDirection.forward);
 
-      expect(viewFocusEvents[1].viewId, view1.viewId);
-      expect(viewFocusEvents[1].state, ui.ViewFocusState.unfocused);
-      expect(viewFocusEvents[1].direction, ui.ViewFocusDirection.undefined);
-
```

Comment:
How come we've lost one event with this change? The listeners are almost the same?

I would expect to have:

* view1 focused
* view2 focused
* view1 unfocused

?

Reply:
Hmmmm maybe I'm missing something; but don't think that's what the browser do. It first calls out the blur events and then the focus one so it would be more like the behavior that's being changed:

1. view 1 focus
2. view 1 unfocus
3. view 2 focus

This is problematic due to the way flutter handles the semantic nodes & text fields. As soon as the user moves out of 
 a text field or semantic node (no matter what's next) it blurs the node and `2)` happens. When `2)` happens it tells the framework to unfocus everything then `3)` happens. When `3` happens, right after, we tell the fraemwork to focus the view back and pick the first focusable widget (node). With this change, even if the input is blurred we let the framework handle the focus traversal by not telling it to unnecesarily focus/unfocus the view.

Reply:
Let me rephrase:

Why did we have 3 events before, and now only 2, if we're using the same event listeners? :P

Reply:
activeElement is either body or null as it transitions across views hence it was interpreted as you were blurring before focusing the view

Reply:
We were previously using `activeElement`. In this change we use the event `target` and `relatedTarget` respectively.